pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -145,6 +145,20 @@ func (p *staticPolicy) validateState(s state.State) error {
 				cID, cset.String(), tmpDefaultCPUset.String())
 		}
 	}
+
+	// 3. It's possible that the topology has changed since the state was written. This can be due to
+	// for example offlining a CPU when kubelet is not running. If this happens, we will run into trouble
+	// when later we try to assign non-existent CPUs to containers. Validate that the topology we got
+	// during startup matches with the set of CPUs we have stored in the state.
+	totalKnownCPUs := tmpDefaultCPUset.Clone()
+	for _, cset := range tmpAssignments {
+		totalKnownCPUs = totalKnownCPUs.Union(cset)
+	}
+	if !totalKnownCPUs.Equals(p.topology.CPUDetails.CPUs()) {
+		return fmt.Errorf("current CPU topology (%s) doesn't match with CPUs in state (%s)",
```

Comment:
"...the CPUs stored in the state..."

Reply:
nit: this has less to do with topology than e.g. "the set of available CPUs". I think adding topology in this error message might confuse a user. nit2: could we format the cpuset so it's surrounded by double-quotes instead of parens? That would match the other logs in this module better IIRC.